Tasks/Kubernetes/src/kubernetes.ts

```diff
@@ -31,47 +32,50 @@ var registryAuthenticationToken = authenticationProvider.getAuthenticationToken(
 
 // open kubectl connection and run the command
 var connection = new ClusterConnection();
-connection.open(tl.getInput("kubernetesServiceEndpoint")).then( 
-    () => run(connection, registryAuthenticationToken),
-    (err) => tl.setResult(tl.TaskResult.Failed, err.message)
-).catch((error) => tl.setResult(tl.TaskResult.Failed, error) );
-
+try
+{
+    connection.open(tl.getInput("kubernetesServiceEndpoint")).then(  
+        () => { return run(connection, registryAuthenticationToken) }
+    ).then(
+       () =>  {
+           tl.setResult(tl.TaskResult.Succeeded, "");
+           connection.close();
+       }
+    ).catch((error) => {
+       tl.setResult(tl.TaskResult.Failed, error.message)
+       connection.close();
+    });
+}
+catch (error)
```

Comment:
Why was this try catch needed? catch at the end of promise should be enough, right?

Reply:
That catches exceptions only in the run function and not in the open function